aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -113,14 +113,16 @@ Tensor isclose(const Tensor& self, const Tensor& other, double rtol, double atol
   // by the default scalar type then this may cause an incorrect result.
 
   // Computes allowed and actual error
-  Tensor cast_other;
+  Tensor cast_self, cast_other;
   if (c10::isIntegralType(self.scalar_type(), /*includeBool=*/true)) {
+    cast_self = self.to(at::get_default_dtype());
```

Comment:
Prior to this change, there was no casting for `self`, but now there is. When `self` and `other` both have bool dtype, casting both makes sense, but do we want to do that when the dtype is not bool?

Reply:
Yeah, for the `self`, It would be automatically casted when subtracting with `casted_other`, so the casting would be redundant here. I've improved the code, please take a look.
https://github.com/pytorch/pytorch/blob/5144381b1d7ce6aebb159615c518b2472136b34c/aten/src/ATen/native/TensorCompare.cpp#L137